src/router/outputqueued/InputQueue.cc

```diff
@@ -159,19 +160,26 @@ void InputQueue::processPipeline() {
     // ensure RFE is empty
     assert(rfe_.flit == nullptr);
 
-    // pull out the front flit
+    // get the front flit
     Flit* flit = buffer_.front();
-    buffer_.pop();
 
-    // send a credit back
-    router_->sendCredit(port_, vc_);
+    // when store and forward is enabled, wait for the whole packet
+    assert(!storeAndForward_ || depth_ >= flit->packet()->numFlits());
+    if (!storeAndForward_ || !flit->isHead() ||
+        buffer_.size() >= flit->packet()->numFlits()) {
```

Comment:
this makes my head hurt. how about writing it as something like

bool routeFlit;
if (storeAndForward) {
  routeFlit = ...;
} else {
  routeFlit = true;
}

since this is how I had to read it anyway

Reply:
Ack